server/src/server.rs

```diff
@@ -1146,12 +1146,17 @@ where
 				RpcServiceCfg::OnlyCalls,
 			));
 
-			Box::pin(
-				http::call_with_service(request, batch_config, max_request_size, rpc_service, max_response_size)
-					.map(Ok),
-			)
+			Box::pin(async move {
+				let rp =
+					http::call_with_service(request, batch_config, max_request_size, rpc_service, max_response_size)
+						.await;
+				drop(conn);
```

Comment:
And this is the main bit; droping here to pull it into the async function after the await.

Perhaps worth a comment to make clear we are dropping it for this reaosn?